paddle/fluid/memory/detail/memory_block.h

```diff
@@ -41,17 +41,10 @@ struct MemoryBlock {
   void init(MetadataCache* cache, Type t, size_t index, size_t size,
             void* left_buddy, void* right_buddy);
 
-  // All these accessors returns fields in the MemoryBlock::Desc of the memory
-  // block.  They all need a MetadataCache instance as their first
-  // parameter because they read the MemoryBlock::Desc from the cache.
-  Type type(const MetadataCache& cache) const;
-  size_t size(const MetadataCache& cache) const;
-  size_t index(const MetadataCache& cache) const;
-  size_t total_size(const MetadataCache& cache) const;
-  bool has_left_buddy(const MetadataCache& cache) const;
-  bool has_right_buddy(const MetadataCache& cache) const;
-  MemoryBlock* left_buddy(const MetadataCache& cache) const;
-  MemoryBlock* right_buddy(const MetadataCache& cache) const;
+  bool get_left_buddy(const MetadataCache& cache,
+                      MemoryBlock*& buddy) const;  // NOLINT
+  bool get_right_buddy(const MetadataCache& cache,
```

Comment:
Do not use non-const reference as output.

Reply:
done